utility/utils.py

```diff
@@ -146,6 +146,128 @@ def fuse_mount(fuse_clients, mounting_dir):
         log.error(e)
 
 
+def test_user_stats_consistency(primary_rgw_node, secondary_rgw_node):
+    """
+    verify and monitor sync consistency via bucket stats across sites.
+    """
+    log.info("Test number of users are consistent across sites")
+    total_users_on_primary = json.loads(
+        primary_rgw_node.exec_command(cmd="sudo radosgw-admin user list | wc -l")[0]
+    )
+    total_users_on_secondary = json.loads(
+        secondary_rgw_node.exec_command(cmd="sudo radosgw-admin user list | wc -l")[0]
+    )
+    user_list_primary = json.loads(
+        primary_rgw_node.exec_command(cmd="sudo radosgw-admin user list")[0]
+    )
+    if total_users_on_primary == total_users_on_secondary:
+        total_user = total_users_on_secondary - 2
+        for user in range(0, total_user):
+            user_name = user_list_primary[user]
+            (
+                tenancy,
+                uid,
+                tenant,
+                pri_size,
+                sec_size,
+                pri_objects,
+                sec_objects,
+            ) = test_tenancy(primary_rgw_node, secondary_rgw_node, user_name)
+            if pri_size == sec_size and pri_objects == sec_objects:
+                log.info(f"user stats are consistent across sites for {user_name}")
+            else:
+                log.info(
+                    "user stats inconsistent, perform sync-stats on both sites and retest for consistency"
+                )
+                if tenancy:
+                    primary_rgw_node.exec_command(
+                        cmd=f"sudo radosgw-admin user stats --sync-stats --uid {uid} --tenant {tenant}"
+                    )
+                    secondary_rgw_node.exec_command(
+                        cmd=f"sudo radosgw-admin user stats --sync-stats --uid {uid} --tenant {tenant}"
+                    )
+                else:
+                    primary_rgw_node.exec_command(
+                        cmd=f"sudo radosgw-admin user stats --sync-stats --uid {user_name}"
+                    )
+                    secondary_rgw_node.exec_command(
+                        cmd=f"sudo radosgw-admin user stats --sync-stats --uid {user_name}"
+                    )
```

Comment:
@viduship , could you shrink the code snippet like this
cmd = f"sudo radosgw-admin user stats --sync-stats --uid {uid}"
if tenancy:
    cmd = f"{cmd} --tenant {tenant}"
primary_rgw_node.exec_command(cmd=cmd)
secondary_rgw_node.exec_command(cmd=cmd)

Reply:
@hmaheswa Addressed.